openedx/core/djangoapps/programs/tests/test_utils.py

```diff
@@ -967,6 +977,147 @@ def test_multiple_published_course_runs(self):
+        """
+        course1 = _create_course(self, self.course_price, course_run_count=2, make_entitlement=True)
+        course2 = _create_course(self, self.course_price, course_run_count=2, make_entitlement=True)
+        expected_skus = set([course1['entitlements'][0]['sku'], course2['entitlements'][0]['sku']])
+        program = ProgramFactory(
+            courses=[course1, course2],
+            is_program_eligible_for_one_click_purchase=True,
+            applicable_seat_types=[CourseMode.VERIFIED],
+        )
+        data = ProgramDataExtender(program, self.user).extend()
+        self.assertTrue(data['is_learner_eligible_for_one_click_purchase'])
+        self.assertEqual(set(data['skus']), expected_skus)
+
+    def test_learner_eligibility_for_one_click_purchase_ineligible_program(self):
+        """
+        Learner should not be eligible for one click purchase if the program is not eligible for one click purchase
+        """
+        course1 = _create_course(self, self.course_price, course_run_count=2, make_entitlement=True)
+        course2 = _create_course(self, self.course_price, course_run_count=2, make_entitlement=True)
+        program = ProgramFactory(
+            courses=[course1, course2],
+            is_program_eligible_for_one_click_purchase=False,
+            applicable_seat_types=[CourseMode.VERIFIED],
+        )
+        data = ProgramDataExtender(program, self.user).extend()
+        self.assertFalse(data['is_learner_eligible_for_one_click_purchase'])
+        self.assertEqual(data['skus'], [])
+
+    def test_learner_eligibility_for_one_click_purchase_user_entitlements(self):
+        """
+        Learner should be eligibile for one click purchase if they hold an entitlement in one or more courses
+        in the program and there are remaining unpurchased courses in the program with entitlement products.
+        """
+        course1 = _create_course(self, self.course_price, course_run_count=2, make_entitlement=True)
+        course2 = _create_course(self, self.course_price, course_run_count=2, make_entitlement=True)
+        CourseEntitlementFactory(user=self.user, course_uuid=course1['uuid'], mode=CourseMode.VERIFIED)
+        expected_skus = set([course2['entitlements'][0]['sku']])
+        program = ProgramFactory(
+            courses=[course1, course2],
+            is_program_eligible_for_one_click_purchase=True,
+            applicable_seat_types=[CourseMode.VERIFIED],
+        )
+        data = ProgramDataExtender(program, self.user).extend()
+        self.assertTrue(data['is_learner_eligible_for_one_click_purchase'])
+        self.assertEqual(set(data['skus']), expected_skus)
+
+    def test_all_courses_owned(self):
+        """
+        Learner should not be eligible for one click purchase if they hold entitlements in all courses in the program.
+        """
+        course1 = _create_course(self, self.course_price)
+        course2 = _create_course(self, self.course_price)
```

Comment:
Please make entitlements for these so that we can be sure that `is_learner_eligible_for_one_click_purchase` is `False` because the user already owns all of the entitlements, not because the courses don't have entitlement products.